DOC-3814: Key concepts - notaries (#543)
corda/corda-docs-portal: content/en/platform/corda/4.8/open-source/key-concepts-notaries.md

## Summary

> * *Notary clusters prevent “double-spends”*

Reviewer: Having now read the full article, I think it would be best to define what a notary is (the concept) vs what the notary cluster is (the physical infrastructure) and use accordingly throughout.

* *Notary clusters are also time-stamping authorities. If a transaction includes a time-window, it can only be notarized during that window*
* *Notary clusters may optionally also validate transactions, in which case they are called “validating” notaries, as opposed to “non-validating”*
* *A network can have several notary clusters*

## Video

{{% vimeo 214138458 %}}

## Overview

> The notary cluster is Corda's uniqueness consensus service. The notary's role is to ensure a

Reviewer: I'd start with a simpler explanation that notaries prevent double spends (and define double spends). 

I also think it's worthwhile to a) link to the consensus key concept doc and b) briefly note that this is a concept unique to corda, because most blockchains reach consensus through broadcasting/byzantine fault tolerance. (i.e. we are not burning the planet!) without going too deep into consensus as it's a different topic.

<a name="key-concepts-notaries-id1"></a>

\[1\]
A state reference is composed of the issuing transaction’s ID and the state’s position in the outputs. It does not
reveal what kind of state it is or its contents.

## Multiple notaries

Each Corda network can have multiple notary clusters. This has several benefits:

> * **Privacy** - with both validating and non-validating notary clusters on the same network, nodes can choose the preferred
> notary cluster on a per-transaction basis.

Reviewer: ```suggestion
* **Privacy** -  Nodes can choose the most suitable notary cluster for a specific transaction.
```

* **Load balancing** - spreading the transaction load over multiple notary clusters allows higher transaction
throughput for the platform overall.

> * **Low latency** - latency can be minimized by choosing a notary cluster physically closer to the transacting parties.

Reviewer: ```suggestion
* **Low latency** - Nodes can speed up transactions by choosing the closest notary cluster to the transacting parties.
```

I'd make this clearer using the active voice and simplifying the language

### Changing notaries

Remember that a notary cluster will only sign a transaction if it is the appointed notary cluster of all the
transaction’s input states. However, there are cases in which it may be necessary to change a state’s appointed notary cluster.

> These include:

Reviewer: ```suggestion
You might need to change a state's appointed notary if:
```

> * When a single transaction needs to consume several states that have different appointed notary clusters
> * When a node would prefer to use a different notary cluster for a given transaction due to privacy or efficiency
> concerns

Reviewer: ```suggestion
* A single transaction needs to consume several states that have different appointed notary clusters.
* A node would prefer to use a different notary cluster for a given transaction for increased privacy or reduced latency.
```

> Before these transactions can be created, the states must first all be re-pointed to the same notary cluster. This is

Reviewer: ```suggestion
Before you create these transactions, you must re-point all the states to the same notary cluster. You can do this 
```

> achieved using a special `notary-change` transaction that takes:

Reviewer: ```suggestion
using a special `notary-change` transaction that takes:
```

* A single input state
* An output state identical to the input state, except that the appointed notary cluster has been changed

> The input state’s appointed notary cluster will sign the transaction if it does not constitute a double-spend, at which
> point a state will enter existence with all the properties of the old state, but with a different appointed notary
> cluster.

Reviewer: ```suggestion
The input state’s appointed notary cluster signs the transaction if it does not constitute a double-spend. The state then has all the properties of the old state, but with a different appointed notary
cluster.
```
